### PR TITLE
GUACAMOLE-736: Explicitly pull in javax.xml.bind depdendency

### PR DESCRIPTION
**Pull request by necouchman**
Updates pom.xml to include the javax.xml.bind dependency explicitly for modules that require it.

### CODE REVIEW

**Comment by necouchman** [comment]
Okay, licenses have been added.  Also bumped version to 2.3.1.  FWIW, it looks like starting with 2.3.2 the license changes to EDL-1.0, but I don't see 2.3.2 out in Maven, so we'll go with 2.3.1 and CDDL for now.

File: extensions/guacamole-auth-cas/src/licenses/LICENSE [hunk 1]
```diff
@@ -234,3 +234,13 @@ JSR-330 / Dependency Injection for Java (http://code.google.com/p/atinject/)
     From: 'JSR-330 Expert Group' (https://jcp.org/en/jsr/detail?id=330)
     License(s):
         Apache v2.0 (bundled/javax.inject-1/LICENSE-2.0.txt)
+
+
+JAXB API / XML Processing (https://projects.eclipse.org/projects/ee4j.jaxb)
```
Comment (mike-jumper) [inline extensions/guacamole-auth-cas/src/licenses/LICENSE]: "JAXB ..." should be between "Google Guice ..." and "JSR-330 ...".

File: extensions/guacamole-auth-duo/src/licenses/LICENSE [hunk 2]
```diff
@@ -272,3 +272,12 @@ JSR-330 / Dependency Injection for Java (http://code.google.com/p/atinject/)
     License(s):
         Apache v2.0 (bundled/javax.inject-1/LICENSE-2.0.txt)
 
+
+JAXB API / XML Processing (https://projects.eclipse.org/projects/ee4j.jaxb)
```
Comment (mike-jumper) [inline extensions/guacamole-auth-duo/src/licenses/LICENSE]: "JAXB ..." should be between "Google Guice ..." and "JSR-330 ...".

File: extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-dist/src/licenses/LICENSE [hunk 3]
```diff
@@ -247,6 +247,15 @@ JSR-330 / Dependency Injection for Java (http://code.google.com/p/atinject/)
         Apache v2.0 (bundled/javax.inject-1/LICENSE-2.0.txt)
 
 
+JAXB API / XML Processing (https://projects.eclipse.org/projects/ee4j.jaxb)
```
Comment (mike-jumper) [inline extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-dist/src/licenses/LICENSE]: "JAXB ..." should be between "Guava ..." and "JSR-330 ...".

File: extensions/guacamole-auth-totp/src/licenses/LICENSE [hunk 4]
```diff
@@ -292,3 +292,12 @@ ZXing Barcode Scanning Library (https://github.com/zxing/zxing/)
     License(s):
         Apache v2.0 (bundled/zxing-3.3.1/LICENSE)
 
+
+JAXB API / XML Processing (https://projects.eclipse.org/projects/ee4j.jaxb)
```
Comment (mike-jumper) [inline extensions/guacamole-auth-totp/src/licenses/LICENSE]: "JAXB ..." should be before "ZXing ...".

**Comment by necouchman** [comment]
Should be properly (re-)ordered, now.

File: extensions/guacamole-auth-totp/src/licenses/LICENSE [hunk 5]
```diff
@@ -284,6 +284,15 @@ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
+JAXB API / XML Processing (https://projects.eclipse.org/projects/ee4j.jaxb)
```
Comment (mike-jumper) [inline extensions/guacamole-auth-totp/src/licenses/LICENSE]: Sorry - I was wrong here. There are a number of other licenses cited in `LICENSE` for TOTP, and the "JAXB ..." one should thus go even further up where it's properly alphabetical.

Comment (necouchman) [reply extensions/guacamole-auth-totp/src/licenses/LICENSE]: Now I've said my A B Cs, next time won't you sing with me... :musical_note:

**Pull request merged**